hathor/mining/block_template.py

```diff
@@ -60,11 +68,25 @@ def generate_mining_block(self, rng: Random, merge_mined: bool = False, address:
         from hathor.transaction.scripts import create_output_script
 
         parents = list(self.get_random_parents(rng))
-        output_script = create_output_script(address) if address is not None else b''
         base_timestamp = timestamp if timestamp is not None else self.timestamp_now
         block_timestamp = min(max(base_timestamp, self.timestamp_min), self.timestamp_max)
-        tx_outputs = [TxOutput(self.reward, output_script)]
-        cls: Union[type['Block'], type['MergeMinedBlock']] = MergeMinedBlock if merge_mined else Block
+        tx_outputs = []
+        if self.reward:
+            output_script = create_output_script(address) if address is not None else b''
+            tx_outputs = [TxOutput(self.reward, output_script)]
+
+        # TODO: Change to a generic function
+        cls: type[Block] | type[MergeMinedBlock] | type[PoaBlock]
```

Comment:
I guess this method should receive the class itself or an enum (instead of `merge_mined: bool`).

Reply:
I agree, that's why I added the `TODO`, but initially I wanted to to it on a separate PR because I thought I would have to change a lot of call sites.

However I checked it and it wasn't a lot, so I did it in https://github.com/HathorNetwork/hathor-core/pull/1061/commits/2885ed533ccfaf10c646950817971e2d6e476350

Reply:
Great! Thanks!